paddle/framework/operator.h

```diff
@@ -116,10 +112,14 @@ class OperatorBase {
   void SetType(const std::string& type) { type_ = type; }
   const AttributeMap& Attrs() const { return attrs_; }
 
+  // Return a new operator instance, which is as same as this.
+  // Use unique_ptr to prevent caller forget to delete this pointer.
+  virtual std::unique_ptr<OperatorBase> Clone() const = 0;
```

Comment:
https://google.github.io/styleguide/cppguide.html#Ownership_and_Smart_Pointers
In google C++ style, says

> Prefer to use std::unique_ptr to make ownership transfer explicit.